### ISSUE TITLE
Segementation fault when saving netcdf with list in attributes

### ISSUE THREAD

**Issue by jws52** [pygeode/pygeode#113]
I found this happens in pygeode1.2.0 (and in 1.1.0) with a dataset or a variable. If I create an attribute where the value is a list, working with it is still fine, but saving to netcdf causes a segmentation fault. Minimal example for a variable:

```
test = pyg.Var(name='testvar',values=range(3),axes=[pyg.Lon(3)]);
test.atts['someitem']='a'
pyg.save('test.nc',test) # this succeeds
test.atts['somelist']=['a']
test.atts['somelist']
pyg.save('test.nc',test) # this causes "Segmentation fault (core dumped)"
```

Perhaps this is an issue with netcdf rather than pygeode, but I'm not sure.

I don't need this resolved, but thought I'd raise it in case it is unintended behaviour.

**Comment by neishm** [comment]
Thanks for reporting this error.  I can reproduce the problem on my end, looking into it now.  In the mean time, saving through the newer "netcdf4" format option seems to work (requires the [netcdf4-python](http://unidata.github.io/netcdf4-python/) package):
```python
pyg.save('test.nc',pyg.Dataset([test]),format='netcdf4')  # No crash
```

**Comment by neishm** [comment]
Looks like this is related to #54.  There was some discussion about supporting lists of variable-length strings (NC_STRING data type), but there were no concrete examples of using this data type in the wild.  The pygeode netcdf module should have detected an unsupported type, emitted a warning, and skipped the attribute.  Unfortunately, this was only detected for strings with > 1 character (e.g. if you set `test.atts['somelist']=['ab']`).  For lists of 1-character strings, it's trying to do something unexpected with a pointer.  As a first step I'll try to fix the 1-character case so it at least gracefully skips the attribute.

**Comment by neishm** [comment]
The segmentation fault should be fixed in the release-1.2 branch now, and will be in the next release (either 1.2.3 or 1.3.0).
